core/new-gui/src/app/workspace/service/dynamic-schema/source-tables/source-tables.service.ts

```diff
@@ -166,27 +164,34 @@ export class SourceTablesService {
   }
 
   private registerUpdateUserFileInFileSourceOp(): void {
-    this.userFileService.getUserFilesChangedEvent().subscribe(_ => {
-      this.userFileNames = this.userFileService.getUserFiles().map(file => `${file.ownerName}/${file.file.name}`);
-
-      Array.from(this.dynamicSchemaService.getDynamicSchemaMap().keys()).forEach(operatorID => {
-        const schema = this.dynamicSchemaService.getDynamicSchema(operatorID);
-        // if operator input attributes are in the result, set them in dynamic schema
-        const fileSchema = this.changeInputToEnumInJsonSchema(
-          schema,
-          fileNameInJsonSchema,
-          this.userFileNames,
-          "File Name"
-        );
-        if (!fileSchema) {
-          return;
-        }
-        if (!isEqual(schema, fileSchema)) {
-          SchemaPropagationService.resetAttributeOfOperator(this.workflowActionService, operatorID);
-          this.dynamicSchemaService.setDynamicSchema(operatorID, fileSchema);
-        }
+    this.userFileService
+      .getUserFilesChangedEvent()
+      .pipe(
+        concatMap(_ => {
+          return this.userFileService.retrieveDashboardUserFileEntryList();
+        })
+      )
+      .subscribe(dashboardFileEntries => {
+        this.userFileNames = dashboardFileEntries.map(file => `${file.ownerName}/${file.file.name}`);
```

Comment:
the source table service updates files whenever the user file changes. however, if user file does not change, how does it initialize the file list?

Reply:
The current workaround I have is the `updateUserFilesChangedEvent()` function I added to the [UserFileService](https://github.com/Texera/texera/pull/1476/files#diff-994ddea3dc54457a32f17d8e67472a79d3cf3d1acdb650b4e9a1e9d54289fd4b).  This function can be invoked to let the UserFileService know about a user file change so it can update `dashboardUserFileEntryChanged`.  

In the original UserFileService, it would indicate a user file changed upon a refresh / clear of the files.  So I updated the [UserFileSection](https://github.com/Texera/texera/pull/1476/files#diff-816ec4819c05d710268d800302852b6a2465fe3acdb38036d03e9a7972279cf6) to invoke `updateUserFilesChangedEvent()` to inform the UserFileService whenever files would be refreshed or cleared.

Would this be a good way to handle the source table service updating files?